lib/datadog/lambda/metrics.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+#
+# Unless explicitly stated otherwise all files in this repository are licensed
+# under the Apache License Version 2.0.
+#
+# This product includes software developed at Datadog (https://www.datadoghq.com/).
+# Copyright 2023 Datadog, Inc.
+#
+
+require 'datadog/statsd'
+
+module Datadog
+  # Metrics module contains the singleton class Client to send custom
+  # metrics to Datadog.
+  module Metrics
+    URL = 'localhost'
+    PORT = '8125'
+    # Client is a singleton class that instantiates a Datadog::Statsd
+    # client to send metrics to the Datadog Extension if present.
+    class Client
+      private_class_method :new
+
+      def self.instance
+        @instance ||= new
```

Comment:
Do you expect race conditions here? Meaning two Ruby threads call `Metrics::Client.instance`?

I ask because, in theoretical parallel execution scenario, you might end with with two parallel calls to `new`, with the last one winning, but the two callers of `Metrics::Client.instance` will receive distinct instances of the `Client` class.

Reply:
Not really. As it is, this should keep a single instance for all subsequent invocations. 

Unless a user creates multiple threads in which they are sending custom metrics. If this is the case, how could this be addressed?

Reply:
Currently `@metrics_client = Metrics::Client.instance` is the first call to `def self.instance`, and it happens the moment `lib/datadog/lambda.rb` is loaded, so I don't think there's concern around when the first call happens.

Reply:
If we changed things, say not calling `Metrics::Client.instance` at `@metrics_client = Metrics::Client.instance` anymore, but letting the user, at any point of their application, make a call that indirectly does `Metrics::Client.instance` for the first time, we could have this issue.

Again, we don't have to worry about it today.

But if it was a concern, we have two options:
* Add a mutex to `def self.instance`.
* Ensure that when users use multiple copies of `new`, everything still works correctly. I believe today this is true, as you can safely have many copies of the statsd client.

Reply:
Saving this for the future! Thank you so much for the really thorough explanation.